pkg/certificates/endpoint.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package certificates
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/peer"
+
+	"storj.io/storj/pkg/certificates/authorizations"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+)
+
+// Endpoint implements pb.CertificatesServer
+type Endpoint struct {
+	log             *zap.Logger
+	ca              *identity.FullCertificateAuthority
+	authorizationDB *authorizations.DB
+	minDifficulty   uint16
+}
+
+// NewCertificatesServer creates a new certificate signing grpc server
+func NewCertificatesServer(log *zap.Logger, ident *identity.FullIdentity, ca *identity.FullCertificateAuthority, authorizationDB *authorizations.DB, minDifficulty uint16) *Endpoint {
```

Comment:
This should be just `NewEndpoint`

Reply:
indeed.. i'm taking another look myself now that I understand the architecture doc a little better.